lib/assets/assets_cache.dart

```diff
@@ -54,22 +54,27 @@ class AssetsCache {
 
   Future<_StringAsset> _readFile(String fileName) async {
     final string = await rootBundle.loadString('assets/$fileName');
-    return _StringAsset()..value = string;
+    return _StringAsset(string);
   }
 
   Future<_BinaryAsset> _readBinary(String fileName) async {
     final data = await rootBundle.load('assets/$fileName');
     final Uint8List list = Uint8List.view(data.buffer);
 
     final bytes = List<int>.from(list);
-    return _BinaryAsset()..value = bytes;
+    return _BinaryAsset(bytes);
   }
 }
 
 class _Asset<T> {
   T value;
+  _Asset(this.value);
 }
 
-class _StringAsset extends _Asset<String> {}
+class _StringAsset extends _Asset<String> {
+  _StringAsset(String value) : super(value);
+}
 
-class _BinaryAsset extends _Asset<List<int>> {}
+class _BinaryAsset extends _Asset<List<int>> {
+  _BinaryAsset(List<int> value) : super(value);
```

Comment:
bits like this can be extracted to a separate PR to master, to reduce the null safety diff. also any API changes we might wanna do to better support null-safety

Reply:
:+1: reducing null safety patchset size is a great idea as it is going to hang being separated for a while.

I'd be happy to target some changes to `master`. Just point me out to places like that. I've got you on this one.

Reply:
#553 has been merged. I've rebased the PR. This doesn't show up in diff as the `flame-engine:upstream` branch wasn't rebased on `develop`.